common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -430,6 +431,41 @@ public MicrosoftStsTokenRequest createRefreshTokenRequest(@NonNull final Abstrac
         return request;
     }
 
+    @Override
+    public MicrosoftStsTokenRequest createRopcTokenRequest(@NonNull final RopcTokenCommandParameters parameters) throws ClientException {
+        final String methodName = ":createPasswordTokenRequest";
+        Logger.verbose(
+                TAG + methodName,
+                "Creating password token request"
+        );
+
+        final MicrosoftStsRopcTokenRequest request = new MicrosoftStsRopcTokenRequest();
+        request.setGrantType(TokenRequest.GrantTypes.PASSWORD);
+
+        request.setUsername(parameters.getUsername());
+        request.setPassword(parameters.getPassword());
+        request.setClaims(parameters.getClaimsRequestJson());
+        request.setClientId(parameters.getClientId());
+        request.setRedirectUri(parameters.getRedirectUri());
+        request.setScope(StringUtil.join(" ", parameters.getScopes()));
+        setTokenRequestCorrelationId(request);
+
+        if (PopAuthenticationSchemeInternal.SCHEME_POP.equals(parameters.getAuthenticationScheme().getName())) {
```

Comment:
Is this supported by ROPC?

Reply:
... not according to the documents - "Always set to BEARER". You should probably fail if POP is returned, or at least make sure we have telemetry for the fact that it is happening.

Reply:
Which documents are these? Can you link them here?

Reply:
Addressed. Removed PoP support for now from code and indicating that even Bearer support is for testing purposes.